ocaml/xapi/xapi_clustering.ml

```diff
@@ -61,17 +61,17 @@ let ip_of_pif (ref,record) =
     }*)
 let assert_pif_prerequisites pif =
   let assert_pif_permaplugged (ref,record) =
-    if not record.API.pIF_disallow_unplug then failwith (Printf.sprintf "PIF %s allows unplug" record.API.pIF_uuid);
-    if not record.pIF_currently_attached then failwith (Printf.sprintf "PIF %s not plugged" record.API.pIF_uuid)
+    if not record.API.pIF_disallow_unplug then
+      raise Api_errors.(Server_error (pif_allows_unplug, [ record.API.pIF_uuid ] ));
+    if not record.pIF_currently_attached then
+      raise Api_errors.(Server_error (required_pif_is_unplugged, [ record.API.pIF_uuid ] ))
   in
   assert_pif_permaplugged pif;
   ignore(ip_of_pif pif)
 
-let handle_error error =
-  (* TODO: replace with API errors? *)
-  match error with
-  | InternalError message -> failwith ("Internal Error: " ^ message)
-  | Unix_error message -> failwith ("Unix Error: " ^ message)
+let handle_error = function
+  | InternalError message -> raise Api_errors.(Server_error (internal_error, [ message ]))
+  | Unix_error message -> raise Api_errors.(Server_error (internal_error, [ "Unix Error: " ^ message ]))
```

Comment:
I think we do not need to explicitly specify `internal_error`, because the previous failwiths will be converted to this automatically. So probably we shouldn't change this block of code